### PR TITLE
Support Jackal ROS packages on Windows

### PR DESCRIPTION
**Pull request by ooeygui**
This change introduces ROS on Windows support for the Jackal codebase.

### CODE REVIEW

File: jackal_base/src/jackal_diagnostic_updater.cpp [hunk 1]
```diff
@@ -227,6 +243,62 @@ void JackalDiagnosticUpdater::navsatCallback(const nmea_msgs::Sentence::ConstPtr
...
+
+  // Get system structure of interface IP addresses.
+  ULONG flags = GAA_FLAG_INCLUDE_PREFIX;
+
+  // check for both IPv4 and IPv6
+  ULONG family = AF_UNSPEC;
+
+  // pre-allocate 15 KB of buffer is sufficient for most computers
+  int length = static_cast<int>(std::ceil(15e3 / sizeof(IP_ADAPTER_ADDRESSES)));
+  auto addresses = std::make_unique<IP_ADAPTER_ADDRESSES[]>(length);
```
Comment (tonybaltovski) [inline jackal_base/src/jackal_diagnostic_updater.cpp:260]: Does `std::make_unique` need C++14 enabled in the CMakeLists.txt?  Melodic/Bionic targets it but Kinetic/Xenial targets C++11.  However, it looks like ros-win is only for Melodic.

File: jackal_base/src/jackal_diagnostic_updater.cpp [hunk 2]
```diff
@@ -227,6 +243,62 @@ void JackalDiagnosticUpdater::navsatCallback(const nmea_msgs::Sentence::ConstPtr
   }
 }
 
+#if defined(_WIN32)
```
Comment (tonybaltovski) [inline jackal_base/src/jackal_diagnostic_updater.cpp]: Any reason for not folding this into the function?

File: jackal_base/src/jackal_diagnostic_updater.cpp (hunk 1 above)
Comment (ooeygui) [reply jackal_base/src/jackal_diagnostic_updater.cpp:260]: @tonybaltovski Would it be possible for you to create a melodic branch? We can then resubmit the pull request.

File: jackal_base/src/jackal_diagnostic_updater.cpp (hunk 2 above)
Comment (ooeygui) [reply jackal_base/src/jackal_diagnostic_updater.cpp]: Let me re-evaluate this. I think having Wifi and networking broken out into a separate ROS package would make it available to other solutions as well. Maybe that would be a longer term fix?

Comment (ooeygui) [reply jackal_base/src/jackal_diagnostic_updater.cpp]: I collapsed it as suggested

File: jackal_base/src/jackal_diagnostic_updater.cpp (hunk 1 above)
Comment (tonybaltovski) [reply jackal_base/src/jackal_diagnostic_updater.cpp:260]: @ooeygui updated to melodic branch.

File: jackal_base/src/jackal_diagnostic_updater.cpp (hunk 2 above)
Comment (tonybaltovski) [reply jackal_base/src/jackal_diagnostic_updater.cpp]: Thanks!  That would be a good long term fix.  I will accept this to get it merged!

**Comment by tonybaltovski** [comment]
Thanks @ooeygui for this exciting PR!

**Pull request merged**